src/Console/Commands/ProfileFromUrl.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace MedianetDev\PConnector\Console\Commands;
+
+use Illuminate\Console\Command;
+
+class ProfileFromUrl extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'pconnector:generate {url : URL api} {profile? : Name of profile}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Generate a PConnector profile';
+
+
+
+    /**
+     * Execute the console command.
+     *
+     * @return bool|null
+     */
+    public function handle()
+    {
+        $url = $this->argument('url');
+
+
+        // Validate the Url
+        if (filter_var($url, FILTER_VALIDATE_URL, FILTER_FLAG_PATH_REQUIRED) === false) {
+            return $this->error('Not a valid URL!');
+        }
+
+        // Publish the config
+        \Artisan::call('vendor:publish', ['--provider' => 'MedianetDev\PConnector\PConnectorServiceProvider', '--tag' => 'config']);
+
+        // Extract profile from url
+        $gateway = parse_url($url);
+
+        $profiles = config('p-connector.profiles');
+        $profile_name = $this->argument('profile') ?? 'profile';
+
+        // Check existing profile name
+        if (isset($profiles[$profile_name])) {
+            return $this->error('Please check your profile name');
+        }
+
+        // Merge profiles
+        $profiles = $profiles +
+        [
+            $profile_name => [
+                'protocol' => $gateway['scheme'] ?? 'http',
+                'host' => $gateway['host'] ?? 'foo.bar',
+                'port' => $gateway['port'] ?? 80,
+                'prefix' => $gateway['path'] ?? '/',
+            ],
+        ];
+
+        // Add profile into config
+        $path = config_path('p-connector.php');
+        config(['p-connector.profiles' => $profiles]);
+
+        // Set new Config
+        if (file_exists($path)) {
+            file_put_contents($path, "<?php \n return \n {$this->var_export_format(config('p-connector'))};");
+        }
```

Comment:
Hi @adelstiti, after testing those lines, it looks like adding the new profile to the existing profile are changing the whole configuration file for the PConnector, what we expect this command to do is to only add the new profile at the end of the configuration profiles array while respecting the PSR-12 standard, and what is happening in the new array of the profile is added but the single quotes ' of the whole file is changed to double quotes ", the code style is a little bit changed and the comments are completely removed from the file, the whole purpose of this PR is to make the development experience (PX) easier for the developer, but removing the comment is a huge downside, maybe we can use a different approach while writing the new profile.